### PR TITLE
Move golden test files into the corresponding API implementation folder.

### PR DESCRIPTION
**Pull request by shifucun**
Changes in API can be located in a folder instead of several places. This also makes the testing and golden file update locally instead of always triggering all golden testing.

- Changed cloud build machine type to N1_HIGHCPU_32 so it can run locally.
- Remove the option to update golden files in Docker mode as it is never used

### CODE REVIEW

File: build/ci/cloudbuild.goldendiff.yaml [hunk 1]
```diff
@@ -40,4 +40,4 @@ options:
   volumes:
     - name: tmp
       path: /tmp
-  machineType: "E2_HIGHCPU_32"
+  machineType: "N1_HIGHCPU_32"
```
Comment (beets) [inline build/ci/cloudbuild.goldendiff.yaml:43]: Could you document these build changes in the PR description?

File: test/e2e/place_statvar_ranking_test.go [hunk 2]
```diff
@@ -198,8 +198,8 @@ func TestChartConfigRankings(t *testing.T) {
 			})
 
 			goldenFile := path.Join(goldenPath, c.goldenFile)
-			if generateGolden {
-				updateGolden(missingRankings, goldenPath, c.goldenFile)
+			if GenerateGolden {
```
Comment (beets) [inline test/e2e/place_statvar_ranking_test.go:201]: Were these other style updates automated? Which tool did you use to generate these? Would be nice to standardize on these so we can keep this in sync.

File: scripts/update_golden.sh [hunk 3]
```diff
@@ -46,13 +46,4 @@ else
     ARG=""
 fi
 
-if [[ $DOCKER == "true" ]]; then
-  DOCKER_BUILDKIT=1 docker build --tag datacommons/mixer-golden-update  -f build/Dockerfile --target golden-update .
```
Comment (beets) [inline scripts/update_golden.sh:50]: could you also document this removal in the PR description?

File: test/e2e/setup.go [hunk 4]
```diff
@@ -40,16 +40,11 @@ import (
...
+	UseMemdb bool
 }
 
-var generateGolden bool
-
-func init() {
-	flag.BoolVar(
-		&generateGolden, "generate_golden", false, "generate golden files")
-}
+var GenerateGolden = os.Getenv("GENERATE_GOLDEN") == "true"
```
Comment (beets) [inline test/e2e/setup.go:47]: Does this need to be unset on exit?

**Review by beets — APPROVED** [review]
thanks for the cleanup!

File: build/ci/cloudbuild.goldendiff.yaml (hunk 1 above)
Comment (shifucun) [reply build/ci/cloudbuild.goldendiff.yaml:43]: Done

File: scripts/update_golden.sh (hunk 3 above)
Comment (shifucun) [reply scripts/update_golden.sh:50]: Done. The docker mode is never used, so removed to skip updating it.

File: test/e2e/place_statvar_ranking_test.go (hunk 2 above)
Comment (shifucun) [reply test/e2e/place_statvar_ranking_test.go:201]: Updated using vscode IDE.

File: test/e2e/setup.go (hunk 4 above)
Comment (shifucun) [reply test/e2e/setup.go:48]: I think setting the env in the code has no effect to the actual environment. The env is set in update_golden.sh and scoped in that file, so would have no effect outside.

**Review by shifucun — COMMENTED** [review]
Thanks for the review!

**Pull request merged**